### PR TITLE
Add navbar template

### PR DESCRIPTION
**Pull request by marwagado**
add navbar template to include in all pages

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -1,14 +1,6 @@
...
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Cafeteria</title>
+<?php
+    require("templates/header.php");
+    require("templates/footer.php");
 
-    <link rel="stylesheet" href="css/main.css">
-</head>
-<body>
+?>
```
Comment (ahmed-fawzy-metwally) [inline index.php]: Great Job
Maybe end these two lines with `?>` that for include header only
and start new `<?php`  that for include footer only
All this as we write the content between the header part and the footer part.
``` php
<?php require("templates/header.php"); ?>
<!-- These part for page content -->
<?php require("templates/footer.php"); ?>
```

File: templates/footer.php [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<script src="js/jquery.js"></script>
+<script src="js/bootstrap.bundle.js"></script>
+<script src="js/main.js"></script>
+</body>
+</html>
```
Comment (Hossam-Tarek) [inline templates/footer.php]: Add a new line at the end of the file.

File: templates/header.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                    <li class="nav-item">
+                        <a class="nav-link" href="#" role="button" aria-haspopup="true" aria-expanded="false">
+                            <img src="1.jpg" width="40" height="40" class="rounded-circle">
+                        </a>
+                    </li>
+                    <li class="nav-item mt-2 nav-link"><?php if(isset($user)) echo "Waheed"; else echo "Admin" 
+                     ?></li>   
+                </ul>
+            </div>
+        </nav>
```
Comment (Hossam-Tarek) [inline templates/header.php]: why are making 8 spaces indentation make it just 4 spaces.

**Review by Hossam-Tarek — CHANGES_REQUESTED** [review]

File: templates/footer.php (hunk 2 above)
Comment (marwagado) [reply templates/footer.php]: ok

**Review by ahmed-fawzy-metwally — APPROVED** [review]
Good Job.

**Review by ahmed-fawzy-metwally — DISMISSED** [review]
Good.
to use a specific stylesheet, script, the title for the page
add this part to the top of page
``` PHP
<?PHP 
      $PAGE_TITLE = "your-page-title";
      $PAGE_STYLESHEETS = "<link rel='stylesheet' href='.....''>";
      $PAGE_SCRIPTS = "<script src='.......'></script>"; 
?>

```

File: templates/header.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<?php
+    function callback($buffer)
+    {
```
Comment (Hossam-Tarek) [inline templates/header.php]: Put the curly bracket at the end of the line of the function

File: templates/header.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        global $PAGE_TITLE, $PAGE_STYLESHEETS, $PAGE_SCRIPTS;
+        $buffer = preg_replace('/(<\s?title\s?>)(.*?)(<\s?\t/itle\s?>)/i', '$1' . $PAGE_TITLE . '$3', $buffer);
```
Comment (Hossam-Tarek) [inline templates/header.php]: In `(.*?)` there is no need for the `?` sign.  And `(<\s?\t/itle\s?>)`  should be modifed to `(</\s*title\s*>)`

File: templates/header.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        $buffer = preg_replace('/%STYLE_SHEETS%/i',$PAGE_STYLESHEETS , $buffer);
+        $buffer = preg_replace('/%PAGE_SCRIPTS%/i',$PAGE_SCRIPTS , $buffer);
+        return $buffer;
```
Comment (Hossam-Tarek) [inline templates/header.php:7]: You should remove the `i` modifier from the pattern

File: templates/header.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    ob_start("callback");
+?> 
+<?php
+$user = "admin";
+?>
```
Comment (Hossam-Tarek) [inline templates/header.php:13]: For what reason this snippet of code

**Review by Hossam-Tarek — APPROVED** [review]
Well done, but changes the things down.

File: templates/footer.php [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<script src="js/jquery.js"></script>
+<script src="js/bootstrap.bundle.js"></script>
+<script src="js/main.js"></script>
```
Comment (Hossam-Tarek) [inline templates/footer.php]: You should remove the main.js line from here

**Review by Hossam-Tarek — CHANGES_REQUESTED** [review]

File: templates/header.php (hunk 4 above)
Comment (marwagado) [reply templates/header.php]: done

File: templates/header.php (hunk 5 above)
Comment (marwagado) [reply templates/header.php]: done

File: templates/header.php (hunk 7 above)
Comment (marwagado) [reply templates/header.php:13]: it will be replaced  with session

File: templates/footer.php (hunk 8 above)
Comment (marwagado) [reply templates/footer.php]: yes 
done

**Review by Hossam-Tarek — APPROVED** [review]

**Pull request merged**